mlserver/parallel/worker.py

```diff
@@ -91,35 +94,50 @@ async def coro_run(self):
                         # and continue
                         continue
 
-                    await self._process_request(request)
-                elif r is self.model_updates._reader:
-                    model_update = self.model_updates.get()
+                    request_task = asyncio.create_task(self._process_request(request))
+                    request_task.add_done_callback(self._request_cb)
+                elif r is self._model_updates._reader:
+                    model_update = self._model_updates.get()
                     # If the queue gets terminated, detect the "sentinel value"
                     # and stop reading
                     if model_update is END_OF_QUEUE:
                         self._active = False
-                        self.model_updates.task_done()
+                        self._model_updates.task_done()
                         return
 
-                    await self._process_model_update(model_update)
-                    self.model_updates.task_done()
+                    update_task = asyncio.create_task(
+                        self._process_model_update(model_update)
+                    )
+                    update_task.add_done_callback(self._update_cb)
```

Comment:
code similar to creating task  and assigning call back for inference. perhaps consider having a helper function?

Reply:
Good point. We could have a `call_with_cb` helper method?